scripts/cfos/make_dataset_data_tables.py

```diff
@@ -87,113 +58,145 @@ def generate_load_table_files(dataset_tables_dict):
     return tsv_filenames_list
 
 
-def create_dataset_tables_dictionary(validated_dataset, dataset_name, dataset_table_names, schema_dict):
-    """For a given dataset, subset/separate input columns into individual table dataframes."""
-
-    # instantiate empty dictionary to capture {table_name: table_metadata_dataframe}
-    dataset_tables_dict = {}
-
-    # create a subset df for each table defined in the chosen dataset
-    for table in dataset_table_names:
-        # get subset dataframe for specific table for given dataset name
-        table_df = dataset_metadata[schema_dict[dataset_name][table]]
-
-        # TODO: edit the value = dataframe for a table to have the entity:table_name_id format
-
-        # add KVP {table name : table_df} entry to all tables dictionary
-        dataset_tables_dict[table] = table_df
-
-    print(f"Success: Dataset has been partitioned into {len(dataset_table_names)} tables: {dataset_table_names}")
-    return dataset_tables_dict
-
-
-def validate_dataset(dataset):
-    """Validate the full dataset metadata with Schema validator."""
-
-    # capture errors from the validation against the pre-defined validations per column
-    errors = schema_validator.validate(dataset, columns=schema_validator.get_column_names())
-    # get indices of the failed rows
-    errors_index_rows = [e.row for e in errors]
+def validate_dataset(dataset, schema_dict, field_dict):
+    validated_dataset = {}
+
+    for table in list(schema_dict.keys()):
+        if table not in list(dataset.keys()):
+            print(f"Failed: Missing required table {table}.")
+            print(dataset.keys())
+            exit()
+
+        errors = dynamically_validate_df(
+            field_dict=field_dict, 
+            fields_to_validate_list=schema_dict[table], 
+            data_df=dataset[table]
+            )
+        # get indices of the failed rows
+        errors_index_rows = [e.row for e in errors]
+
+        # if any errors, print error message and exit
+        if errors_index_rows:
+            print(f"Failed: Dataset validation errors found in table {table}. Please retry after correcting errors listed in validation_errors.csv.")
+            pd.DataFrame({'validation_error':errors}).to_csv('validation_errors.csv')
+            exit()
+
+        # TODO: determine how to handle drops of failed rows/columns to get cleaned data
```

Comment:
According to [this](https://pandas.pydata.org/docs/reference/api/pandas.read_excel.html) documentation, it looks like pandas.read_excel reads multiple sheets into a dictionary of dataframes. 

The validation code loops over each "sheet" (sheet = table, they're the same thing) and passes in the corresponding dataframe to validate.py for validation.

Reply:
Regarding dataset being the same as validated_dataset, I thought so too, but I think they were treated differently in the initial code and I wasn't sure if it'd impact later code. 

In the previous code, since it was evaluating each table separately, it needed to join all the tables together as they were marked valid. Here, we already have a dict of dataframes, and it may not be necessary anymore. 

I'm not sure if there's code downstream of this that could be consolidated too, I tried to keep my changes relatively contained to the validation logic itself and the reading of the excel.